Problem compiling on ARM

Hello I was trying to compile CNATS for a specific ubuntu 14 arm64 box and when I get to the make install part I keep getting this error:

[  1%] Building C object src/CMakeFiles/nats.dir/unix/mutex.c.o
/tmp/ccjNLaiT.s: Assembler messages:
/tmp/ccjNLaiT.s:176: Error: unknown mnemonic `rep' -- `rep'
src/CMakeFiles/nats.dir/build.make:542: recipe for target 'src/CMakeFiles/nats.dir/unix/mutex.c.o' failed
make[2]: *** [src/CMakeFiles/nats.dir/unix/mutex.c.o] Error 1
CMakeFiles/Makefile2:983: recipe for target 'src/CMakeFiles/nats.dir/all' failed
make[1]: *** [src/CMakeFiles/nats.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

I wonder if any of you has any Idea how to fix this. I was thinking of making my own simple makefile but I need to figure out how to handle forward declarations.

I also tried to compile it in another consumer arm 64 box and it works.

--- a/src/unix/mutex.c
+++ b/src/unix/mutex.c
@@ -60,7 +60,7 @@ void
 natsMutex_Lock(natsMutex *m)
 {
     // The "rep" instruction used for spinning is not supported on ARM.
-#ifndef __arm__
+#if !defined(__arm__) && !defined(__aarch64__)
     if (gLockSpinCount > 0)
     {
         int64_t attempts = 0;

